### PR TITLE
Sprint 2

### PR DESCRIPTION
**Pull request by iAdityaRepo**
Implementing Restaurant Service

### CODE REVIEW

File: restaurants/src/main/java/com/restaurants/controller/FoodItemController.java [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+import java.util.List;
+
+/**
+ * Controller for managing food items in a restaurant.
+ * Provides endpoints for food item management operations including retrieval, creation, and update.
+ */
+@Slf4j
+@RestController
+@RequestMapping("/foodItem")
+@CrossOrigin(origins = "http://localhost:3000") // Adjust this to your frontend URL
```
Comment (Rashi-Choudhary-NucleusTeq) [inline restaurants/src/main/java/com/restaurants/controller/FoodItemController.java]: Remove this comments

File: restaurants/src/main/java/com/restaurants/controller/FoodItemController.java [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+   *
+   * @param multipartFile the file associated with the food item
+   * @return a response entity with the result of the operation
+   */
+//  @PostMapping("/add")
+//  public ResponseEntity<?> addFoodItem(@RequestPart("foodItemInDto") FoodItemInDto foodItemInDto,
+//                                       @RequestPart("multipartFile") MultipartFile multipartFile) {
+//    log.info("Adding new food item: {}", foodItemInDto);
+//    FoodItemOutDto foodItemOutDto = foodItemService.add(foodItemInDto, multipartFile);
+//    log.info("Added food item: {}", foodItemOutDto);
```
Comment (Rashi-Choudhary-NucleusTeq) [inline restaurants/src/main/java/com/restaurants/controller/FoodItemController.java]: Remove this extra commented code before the branch get merged into feature branch.

File: restaurants/src/main/java/com/restaurants/controller/FoodItemController.java [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+//    return new ResponseEntity<>(foodItemOutDto, HttpStatus.CREATED);
+//  }
+  @PostMapping("/addFoodItem")
+  public ResponseEntity<?> addFoodItem(
+    @RequestParam("foodName") String foodName,
+    @RequestParam("restaurantId") Integer restaurantId,
+    @RequestParam("description") String description,
```
Comment (Rashi-Choudhary-NucleusTeq) [inline restaurants/src/main/java/com/restaurants/controller/FoodItemController.java]: Why we are not using indto of food item to get all this data and taking all the inputs in RequestParam

File: restaurants/src/main/java/com/restaurants/controller/FoodItemController.java [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+    @RequestParam("categoryId") Integer categoryId,
+    @RequestParam("isAvailable") Boolean isAvailable,
+    @RequestParam("price") Double price,
+    @RequestParam(value = "multipartFile", required = false) MultipartFile multipartFile) {
+
+    // Create a FoodItemInDto object
+    FoodItemInDto foodItemInDto = new FoodItemInDto();
+    foodItemInDto.setFoodName(foodName);
```
Comment (Rashi-Choudhary-NucleusTeq) [inline restaurants/src/main/java/com/restaurants/controller/FoodItemController.java]: this code should be in service why we have implemented it in controller.

File: restaurants/src/main/java/com/restaurants/controller/RestaurantController.java [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+//    }
+//    log.info("Successfully added restaurant: {}", restaurantOutDto);
+//    return new ResponseEntity<>(restaurantOutDto, HttpStatus.CREATED);
+//  }
+  @PostMapping("/add")
+  public ResponseEntity<?> addRestaurant(
+    @RequestParam("userId") Integer userId,
+    @RequestParam("restaurantName") String restaurantName,
+    @RequestParam("address") String address,
+    @RequestParam("contactNumber") String contactNumber,
```
Comment (Rashi-Choudhary-NucleusTeq) [inline restaurants/src/main/java/com/restaurants/controller/RestaurantController.java:59]: Again, why we are not using indto here

File: restaurants/src/main/java/com/restaurants/dtoconversion/DtoConversion.java [hunk 6]
```diff
@@ -0,0 +1,141 @@
...
+ * Utility class for converting between DTOs and entities.
+ * Provides static methods to map between DTOs and entity objects for Restaurant, Category, and FoodItem.
+ */
+@Slf4j
+public final class DtoConversion {
+  // Private constructor to prevent instantiation
+  private DtoConversion() {
+
+  }
+  public static Restaurant mapToRestaurant(RestaurantInDto restaurantInDto) {
```
Comment (Rashi-Choudhary-NucleusTeq) [inline restaurants/src/main/java/com/restaurants/dtoconversion/DtoConversion.java:25]: javadoc should not be inside the method

File: restaurants/src/main/java/com/restaurants/indto/CategoryInDto.java [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+   * </p>
+   */
+  private Integer restaurantId;
+  /**
+   * The name of the category.
+   * <p>
+   * This field represents the name of the category that will be created or updated.
+   * </p>
+   */
+  private String name;
```
Comment (Rashi-Choudhary-NucleusTeq) [inline restaurants/src/main/java/com/restaurants/indto/CategoryInDto.java:35]: Add some regex to restrict user to give meaningful name to category

File: restaurants/src/main/java/com/restaurants/indto/CategoryInDto.java [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+ */
+@Data
+public class CategoryInDto {
+  /**
+   * The ID of the restaurant to which this category belongs.
+   * <p>
+   * This field is used to associate the category with a specific restaurant.
+   * </p>
+   */
+  private Integer restaurantId;
```
Comment (Rashi-Choudhary-NucleusTeq) [inline restaurants/src/main/java/com/restaurants/indto/CategoryInDto.java:25]: Add not null contraint

File: restaurants/src/main/java/com/restaurants/repository/CategoryRepo.java [hunk 9]
```diff
@@ -0,0 +1,16 @@
+package com.restaurants.repository;
+
+import com.restaurants.entities.Category;
+import org.springframework.data.jpa.repository.JpaRepository;
+
+import java.util.List;
+
+public interface CategoryRepo extends JpaRepository<Category, Integer> {
```
Comment (Rashi-Choudhary-NucleusTeq) [inline restaurants/src/main/java/com/restaurants/repository/CategoryRepo.java:9]: do not use short form, write Repository

File: restaurants/src/main/java/com/restaurants/service/CategoryService.java [hunk 10]
```diff
@@ -0,0 +1,78 @@
...
+import java.util.Optional;
+
+@Slf4j
+@Service
+public class CategoryService {
+  @Autowired
+  private CategoryRepo categoryRepo;
+
+  /**
+   * Add a new category.
```
Comment (Rashi-Choudhary-NucleusTeq) [inline restaurants/src/main/java/com/restaurants/service/CategoryService.java:31]: Add proper javadocs

File: restaurants/src/main/java/com/restaurants/service/CategoryService.java [hunk 11]
```diff
@@ -0,0 +1,78 @@
...
+   */
+  public CategoryOutDto addCategory(CategoryInDto categoryInDto) {
+    log.info("Adding a new category with name: {}", categoryInDto.getName());
+    Category category = categoryRepo.save(DtoConversion.mapToCategory(categoryInDto));
+    log.info("Category added successfully");
+    return DtoConversion.mapToCategoryOutDto(category);
+  }
+
+  /**
+   * Get a category by ID.
```
Comment (Rashi-Choudhary-NucleusTeq) [inline restaurants/src/main/java/com/restaurants/service/CategoryService.java:50]: Add proper javadocs

File: restaurants/src/main/java/com/restaurants/service/FoodItemService.java [hunk 12]
```diff
@@ -0,0 +1,131 @@
...
+  private RestaurantRepo restaurantRepo;
+
+  @Autowired
+  private CategoryRepo categoryRepo;
+
+  /**
+   * Get restaurant name by FoodItem.
+   */
+  public String getRestaurantName(FoodItem foodItem) {
+    log.info("Fetching restaurant name for restaurant ID: {}", foodItem.getRestaurantId());
```
Comment (Rashi-Choudhary-NucleusTeq) [inline restaurants/src/main/java/com/restaurants/service/FoodItemService.java:43]: add proper javadocs

File: restaurants/src/main/java/com/restaurants/util/WebConfig.java [hunk 13]
```diff
@@ -0,0 +1,32 @@
...
+   * Configures CORS mappings for the application.
+   * <p>
+   * This method allows all endpoints to be accessed from the specified origin,
+   * and permits the specified HTTP methods and headers.
+   * </p>
+   *
+   * @param registry the CORS registry to be configured
+   */
+  @Override
+  public void addCorsMappings(CorsRegistry registry) {
```
Comment (Rashi-Choudhary-NucleusTeq) [inline restaurants/src/main/java/com/restaurants/util/WebConfig.java:26]: why we created this method

File: restaurants/src/test/java/com/restaurants/RestaurantsApplicationTests.java [hunk 14]
```diff
@@ -2,12 +2,27 @@
...
+
+  @Autowired
+  private ApplicationContext applicationContext;
 
   @Test
   void contextLoads() {
+    assertNotNull(applicationContext);
   }
 
+//  @Test
```
Comment (Rashi-Choudhary-NucleusTeq) [inline restaurants/src/test/java/com/restaurants/RestaurantsApplicationTests.java]: Remove commented code

File: restaurants/src/test/java/com/restaurants/indto/RestaurantInDtoTest.java [hunk 15]
```diff
@@ -0,0 +1,142 @@
...
+import static org.junit.jupiter.api.Assertions.*;
+
+public class RestaurantInDtoTest {
+
+  private RestaurantInDto restaurantInDto;
+
+  @BeforeEach
+  public void setUp() {
+    restaurantInDto = new RestaurantInDto();
+    // Initialize the object with valid data
```
Comment (Rashi-Choudhary-NucleusTeq) [inline restaurants/src/test/java/com/restaurants/indto/RestaurantInDtoTest.java]: remove this comments

File: user/src/main/java/com/user/util/WebConfig.java [hunk 16]
```diff
@@ -4,13 +4,28 @@
...
+   * This method allows all endpoints to be accessed from the specified origin,
+   * and permits the specified HTTP methods and headers.
+   * </p>
+   *
+   * @param registry the CORS registry to be configured
+   */
   @Override
   public void addCorsMappings(CorsRegistry registry) {
     registry.addMapping("/**")  // Allow all endpoints
-      .allowedOrigins("http://localhost:3000")  // Allow this origin
```
Comment (Rashi-Choudhary-NucleusTeq) [inline user/src/main/java/com/user/util/WebConfig.java:13]: why we replaces 3000 with *

**Review by Rashi-Choudhary-NucleusTeq — CHANGES_REQUESTED** [review]

File: restaurants/src/main/java/com/restaurants/controller/FoodItemController.java (hunk 1 above)
Comment (iAdityaRepo) [reply restaurants/src/main/java/com/restaurants/controller/FoodItemController.java]: done

File: restaurants/src/main/java/com/restaurants/controller/FoodItemController.java (hunk 2 above)
Comment (iAdityaRepo) [reply restaurants/src/main/java/com/restaurants/controller/FoodItemController.java]: done

File: restaurants/src/main/java/com/restaurants/controller/FoodItemController.java (hunk 3 above)
Comment (iAdityaRepo) [reply restaurants/src/main/java/com/restaurants/controller/FoodItemController.java]: now using @ModelAttribute

File: restaurants/src/main/java/com/restaurants/controller/FoodItemController.java (hunk 4 above)
Comment (iAdityaRepo) [reply restaurants/src/main/java/com/restaurants/controller/FoodItemController.java]: resolved

File: restaurants/src/main/java/com/restaurants/util/WebConfig.java (hunk 13 above)
Comment (iAdityaRepo) [reply restaurants/src/main/java/com/restaurants/util/WebConfig.java:26]: for providing the configuring with front end and providing the matcher for backend end points

File: restaurants/src/test/java/com/restaurants/RestaurantsApplicationTests.java (hunk 14 above)
Comment (iAdityaRepo) [reply restaurants/src/test/java/com/restaurants/RestaurantsApplicationTests.java]: done

File: restaurants/src/test/java/com/restaurants/indto/RestaurantInDtoTest.java (hunk 15 above)
Comment (iAdityaRepo) [reply restaurants/src/test/java/com/restaurants/indto/RestaurantInDtoTest.java]: done

File: user/src/main/java/com/user/util/WebConfig.java (hunk 16 above)
Comment (iAdityaRepo) [reply user/src/main/java/com/user/util/WebConfig.java:13]: */
  @Override
  public void addCorsMappings(CorsRegistry registry) {
    registry.addMapping("/**")  
      .allowedOrigins("*")  
      .allowedMethods("*")
      .allowedHeaders("*");
  }

**Comment by iAdityaRepo** [comment]
i make the suggested comments and make the changes

**Pull request merged**